src/Core/Domain/Carrier/Command/SetCarrierRangesCommand.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierId;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierRangesCollection;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+/**
+ * Command aim to edit carrier range
+ */
+class SetCarrierRangesCommand
+{
+    private CarrierId $carrierId;
+
+    public function __construct(
+        int $carrierId,
+        private readonly CarrierRangesCollection $ranges,
```

Comment:
The input parameters of the CQRS command must be scalar/native ones (so only int, string, bool, flattened arrays) The only exceptions to this are:
- `DecimalNumber`
- `ShopConstraint`

So you should pass the flatten array in the constructor and convert it into a `CarrierRangesCollection` right away, this is important especially for the API because it only serializes flatten data as inputs, and it's then the responsibility of the CQRS class to reshape them (we have the same principle with VOs)